epochStart/metachain/auctionListSelector.go

```diff
@@ -86,167 +176,252 @@ func (als *auctionListSelector) SelectNodesFromAuctionList(validatorsInfoMap sta
 		return nil
 	}
 
-	auctionListSize := uint32(len(auctionList))
-	log.Info("systemSCProcessor.SelectNodesFromAuctionList",
+	log.Info("auctionListSelector.SelectNodesFromAuctionList",
 		"max nodes", maxNumNodes,
 		"current number of validators", currNumOfValidators,
 		"num of nodes which will be shuffled out", numOfShuffledNodes,
 		"num of validators after shuffling", numOfValidatorsAfterShuffling,
 		"auction list size", auctionListSize,
-		fmt.Sprintf("available slots (%v -%v)", maxNumNodes, numOfValidatorsAfterShuffling), availableSlots,
+		fmt.Sprintf("available slots (%v - %v)", maxNumNodes, numOfValidatorsAfterShuffling), availableSlots,
 	)
 
-	err = als.sortAuctionList(auctionList, randomness)
-	if err != nil {
-		return err
-	}
-
+	als.displayOwnersData(ownersData)
 	numOfAvailableNodeSlots := core.MinUint32(auctionListSize, availableSlots)
-	als.displayAuctionList(auctionList, numOfAvailableNodeSlots)
-
-	for i := uint32(0); i < numOfAvailableNodeSlots; i++ {
-		newNode := auctionList[i]
-		newNode.SetList(string(common.SelectedFromAuctionList))
-		err = validatorsInfoMap.Replace(auctionList[i], newNode)
-		if err != nil {
-			return err
-		}
-	}
 
-	return nil
-}
+	sw := core.NewStopWatch()
+	sw.Start("auctionListSelector.sortAuctionList")
+	defer func() {
+		sw.Stop("auctionListSelector.sortAuctionList")
+		log.Debug("time measurements", sw.GetMeasurements()...)
+	}()
 
-// TODO: Move this in elrond-go-core
-func safeSub(a, b uint32) (uint32, error) {
-	if a < b {
-		return 0, epochStart.ErrUint32SubtractionOverflow
-	}
-	return a - b, nil
+	return als.sortAuctionList(ownersData, numOfAvailableNodeSlots, validatorsInfoMap, randomness)
 }
 
-func getAuctionListAndNumOfValidators(validatorsInfoMap state.ShardValidatorsInfoMapHandler) ([]state.ValidatorInfoHandler, uint32) {
-	auctionList := make([]state.ValidatorInfoHandler, 0)
+func (als *auctionListSelector) getAuctionDataAndNumOfValidators(
+	validatorsInfoMap state.ShardValidatorsInfoMapHandler,
+	unqualifiedOwners map[string]struct{},
+) (map[string]*ownerData, uint32, uint32, error) {
```

Comment:
make a structure please instead of 3 returns and the error. So one struct and the error.

Reply:
This is now obsolete, since this func is removed in future refactor PR